### PR TITLE
[feat]: [CI-5000]: Resolve Hosted vm secrets on ci-manager

### PR DESCRIPTION
**Pull request by shubham149**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35530)

### CODE REVIEW

File: 332-ci-manager/service/src/test/java/io/harness/ci/CIExecutionTestModule.java [hunk 1]
```diff
@@ -85,6 +88,12 @@ String serviceSecret() {
     return "j6ErHMBlC2dn6WctNQKt0xfyo_PZuK7ls0Z4d6XCaBg";
   }
 
+  @Provides
+  @Named("secretCache")
```
Comment (aman-harness) [inline 332-ci-manager/service/src/test/java/io/harness/ci/CIExecutionTestModule.java:92]: This string is hardcoded at couple of places. Can you move it to a common constant?

**Review by shubham149 — COMMENTED** [review]
> Reviewed 11 of 15 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35530)* status: 15 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`332-ci-manager/service/src/test/java/io/harness/ci/CIExecutionTestModule.java` line 92 at r1](https://reviewable.io/reviews/harness/harness-core/35530#-N7ZQaaJNOc-yGdLefq4-r1-92:-N7_DxWjBc-v8K6rC4k0:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/ce079158fe15e1798e0a5c43b12e71112e092a49/332-ci-manager/service/src/test/java/io/harness/ci/CIExecutionTestModule.java#L92)):*
<details><summary><i>Previously, aman-harness (Aman Singh) wrote…</i></summary><blockquote>

This string is hardcoded at couple of places. Can you move it to a common constant?
</blockquote></details>

Done.

**Review by shubham149 — COMMENTED** [review]
Added the logs in HostedVmSecretManagerFunctor to capture whether secret is decrypted from cache or not.

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35530#-:-N7_E7K51epVmt5nIy7Q:bmp5rju)* status: 15 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 11 of 15 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35530#-:-N7_EVdf92sAT6YXJJUt:bnfp4nl)* status: :shipit: complete! ✓

File: 310-ci-manager/src/main/java/io/harness/app/CICacheRegistrar.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.MapBinder;
+import com.google.inject.name.Named;
+import com.google.inject.name.Names;
+import javax.cache.Cache;
+import javax.cache.expiry.CreatedExpiryPolicy;
+
+@OwnedBy(PL)
```
Comment (harshjain12) [inline 310-ci-manager/src/main/java/io/harness/app/CICacheRegistrar.java:30]: wrong annotation

Comment (shubham149) [reply 310-ci-manager/src/main/java/io/harness/app/CICacheRegistrar.java:30]: Will fix it in next PR

**Pull request merged**